src/test_l1_dsVideoDevice.c

```diff
@@ -1367,34 +1374,35 @@ void test_l1_dsVideoDevice_positive_dsGetCurrentDisplayframerate(void)
     UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
     // Define variables
     int result;
-    int index = 0;
     intptr_t handle = -1;
-    char fetchedFramerate1[50]; // Assuming a buffer size, modify as necessary.
-    char fetchedFramerate2[50]; // Assuming a buffer size, modify as necessary.
+    char fetchedFramerate[50]; // Assuming a buffer size, modify as necessary.
 
     // Step 01: Initialize video devices
     result = dsVideoDeviceInit();
-    UT_ASSERT_EQUAL(result, dsERR_NONE);
+    UT_ASSERT_EQUAL_FATAL(result, dsERR_NONE);
 
     // Step 02: Obtain video device handle
-    result = dsGetVideoDevice(index, &handle);
-    UT_ASSERT_EQUAL(result, dsERR_NONE);
-    UT_ASSERT_TRUE(handle >= 0);
-
-    // Step 03: Get the current display framerate using the obtained handle
-    result = dsGetCurrentDisplayframerate(handle, fetchedFramerate1);
-    UT_ASSERT_EQUAL(result, dsERR_NONE);
-
-    // Step 04: Get the current display framerate again
-    result = dsGetCurrentDisplayframerate(handle, fetchedFramerate2);
-    UT_ASSERT_EQUAL(result, dsERR_NONE);
-
-    // Step 05: Compare the returned values
-    UT_ASSERT_EQUAL(strcmp(fetchedFramerate1, fetchedFramerate2), 0); // They should be the same
-
-    // Step 06: De-initialize the video devices
+    for(int i = 0; i < gDSvideoDevice_NumVideoDevices; i++){
+        result = dsGetVideoDevice(i, &handle);
+        UT_ASSERT_EQUAL(result, dsERR_NONE);
+        UT_ASSERT_NOT_EQUAL(handle, (intptr_t)NULL);
+
+        // Step 03: Get the current display framerate using the obtained handle
+        for (int j=0;j<gDSVideoDeviceConfiguration[i].NoOfSupportedDFR;j++){
+            result = dsGetCurrentDisplayframerate(handle, fetchedFramerate);
+            if(gSourceType == 0) {
+                UT_ASSERT_EQUAL(result, dsERR_NONE);
+
+                // Step 04: Compare the current display framerate with the value from the profile
+	        UT_ASSERT_EQUAL(gDSVideoDeviceConfiguration[i].SupportedDisplayFramerate, fetchedFramerate);
```

Comment:
please check the logic it will be same as dsGetDFC()

Reply:
done